elephant_vending_machine/libraries/vending_machine.py

```diff
@@ -0,0 +1,181 @@
+
+    This class provides an abstraction of the overall machine, exposing SensorGrouping attributes
+    for control of individual groupings of screen/sensor/LED strip.
+
+    Parameters:
+        addresses (list): A list of local IP addresses of the Raspberry Pis
+        config (dict): A dictionary with configuration values, should contain
+            REMOTE_LED_SCRIPT_DIRECTORY: a string representing the absolute path
+            to the directory on the remote pis where the LED scripts are stored,
+            REMOTE_IMAGE_DIRECTORY: a string representing the absolute path
+            to where stimuli images are stored on the remote pis, SENSOR_THRESHOLD:
+            the minimum sensor reading that will not count as motion detected.
+            LEFT_SENSOR_PIN: an integer in the range 0-5 indicating which pin on
+            the maestro board the left sensor pin is wired to. There will also be
+            MIDDLE_SENSOR_PIN and RIGHT_SENSOR_PIN, with corresponding purposes.
+            In the event these values are not passed in, defaults will be assigned
+            as a fallback.
+    """
+
+    def __init__(self, addresses, config=None):
+        self.addresses = addresses
+        if config is None:
+            self.config = {}
+        else:
+            self.config = config
+        if 'REMOTE_IMAGE_DIRECTORY' not in self.config:
+            self.config['REMOTE_IMAGE_DIRECTORY'] = '/home/pi/elephant_vending_machine/images'
+        if 'REMOTE_LED_SCRIPT_DIRECTORY' not in self.config:
+            self.config['REMOTE_LED_SCRIPT_DIRECTORY'] = '/home/pi/rpi_ws281x/python'
+        if 'LEFT_SENSOR_PIN' not in self.config:
+            self.config['LEFT_SENSOR_PIN'] = LEFT_SENSOR_PIN
+        if 'MIDDLE_SENSOR_PIN' not in self.config:
+            self.config['MIDDLE_SENSOR_PIN'] = MIDDLE_SENSOR_PIN
+        if 'RIGHT_SENSOR_PIN' not in self.config:
+            self.config['RIGHT_SENSOR_PIN'] = RIGHT_SENSOR_PIN
+        if 'SENSOR_THRESHOLD' not in self.config:
+            self.config['SENSOR_THRESHOLD'] = SENSOR_THRESHOLD
+        self.left_group = SensorGrouping(
+            addresses[0], LEFT_SCREEN, self.config['LEFT_SENSOR_PIN'], self.config)
+        self.middle_group = SensorGrouping(
+            addresses[1], MIDDLE_SCREEN, self.config['MIDDLE_SENSOR_PIN'], self.config)
+        self.right_group = SensorGrouping(
+            addresses[2], RIGHT_SCREEN, self.config['RIGHT_SENSOR_PIN'], self.config)
+        self.result = None
+
+    @staticmethod
+    def wait_for_input(groups, timeout):
+        """Waits for input on the motion sensors. If no motion is detected by the specified
+        time to wait, returns with a result to indicate this.
+
+        Parameters:
+            groups (list[SensorGrouping]): The SensorGroupings which should be monitored for input.
+            timeout (int): The amount of time in seconds to wait for input before timing out and
```

Comment:
Should check with Katie that seconds are granular enough for her use case. Otherwise we can use milliseconds.

Reply:
I'll ask her Monday

Reply:
This has been addressed as of commit c464fdf